core/src/mindustry/world/Block.java

```diff
@@ -106,6 +106,8 @@ public class Block extends UnlockableContent implements Senseable{
     public boolean underBullets;
     /** whether this is rotatable */
     public boolean rotate;
+    /** if rotate is false and this is true, building rotation is set to 0 regardless of player input */
+    public boolean lockRotation = true;
```

Comment:
Should it default to false?

Reply:
probably not, PRed values probably shouldnt force changes that may affect vanilla if intentionally avoidable